lib/storage.js

```diff
@@ -0,0 +1,61 @@
+'use strict';
+
+const Promise = require('bluebird');
+const createError = require('http-errors');
+const debug = require('debug')('athletesApp: storage');
+const fs = Promise.promisifyAll(require('fs'), {suffix: 'Prom'});
+
+const mkdirp = Promise.promisifyAll(require('mkdirp'));
+const storage = module.exports =  {};
+
+
+// create item
+// will reject a 400 error if there is no schema or item
+storage.createItem = function(schemaName, item){
+  debug('createItem()');
+  if (!schemaName) return Promise.reject(createError(400, 'expected schemaName'));
+  if (!item) return Promise.reject(createError(400, 'expected item'));
+  if (!item.id) return Promise.reject(createError(400, 'expected item to have an id'));
+
+  let json = JSON.stringify(item);
+  let path = `./data/${schemaName}`;
+  return fs.accessProm(path)
+  .catch(err => {
+    if (err.code === 'ENOENT')
+      return mkdirp.mkdirpAsync(path);
+    else
+      return Promise.reject(err);
+  })
+  .then(() => fs.writeFileProm(`${path}/${item.id}.json`, json))
+  .then(() => item);
+};
+
+storage.fetchItem = function(schemaName, id){
+  debug('fetchItem()');
+  if (!schemaName) return Promise.reject(createError(400, 'expected schemaName'));
+  if (!id) return Promise.reject(createError(400, 'expected item ID'));
+  let path = `./data/${schemaName}`;
+  return fs.readFileProm(`${path}/${id}.json`)
+  .catch(err => Promise.reject(createError(404, err.message)))
+  .then(data => {
+    let item = JSON.parse(data.toString());
+    return item;
+  });
+};
+
+storage.deleteItem = function(schemaName, id){
+  debug('deleteItem()');
+  if (!schemaName) return Promise.reject(createError(400, 'expected schemaName'));
+  if (!id) return Promise.reject(createError(400, 'expected item'));
+  let path = `./data/${schemaName}`;
+  return fs.unlinkProm(`${path}/${id}.json`)
+  .catch(err => Promise.reject(createError(404, err.message)));
+};
+
+storage.availIDs = function(schemaName) {
+  debug('availIDs()');
+  let path = `./data/${schemaName}`;
+  return fs.readdirProm(`${path}`)
+  .catch(err => Promise.reject(createError(404, err.message)))
+  .then( filenames => filenames.map(name => name.split('.json')[0]));
+};
```

Comment:
Copy/pasted from a comment I left for Jaren:

Good job with the bonus. I wrote a way you can get rid of ``.gitignore`` from the array if you want to utilize it (because from an API requester's standpoint, that's a little weird to see that filename returned to them). 

I don't know if you remember from 301, but you can chain array methods to each other, so I just chained an additional ``filter`` method to take out ``.gitignore`` and just return ID numbers. I hope it helps!

```
storage.availableIDs = function(schema) {
  debug('availableIDs()');
  return fs.readdirProm(`${dataDir}/${schema}`)
  .catch(err => Promise.reject(createError(404, err.message)))
  .then(filenames => {
    // filenames = filenames.map(name => name.split('.json')[0]);
    filenames = filenames.map((name) => {
      name = name.split('.json')[0];
      return name;
    }).filter((element) => {
      return element !== '.gitignore';
    });
    return filenames;
  });
};
```